### PR TITLE
updates CSV : Deploy Local-Storage-Operator

### PR DESCRIPTION
**Pull request by umangachapagain**
- Adds local-storage-operator as dependency of ocs-operator
- Updates deploy-with-olm.yaml to include local-storage-operator

Signed-off-by: Umanga Chapagain <chapagainumanga@gmail.com>

-----
Few things to note,
1. `local-storage-operator` needs to be a part of `olm-catalog` for dependency to be resolved automatically.
2. `deploy-with-olm.yaml` deploys the catalogSource and operatorGroup for `local-storage-operator` along with `ocs-operator`.
3. To be able to test this, we need a new build of `bundle` at `quay.io/ocs-dev/ocs-registry`

### CODE REVIEW

**Comment by kshlm** [comment]
I know this a WIP. But please keep the Rook and NooBaa CRDs in the owned list. The OCS CSV deploys these CRDs and the operators itself. If we move these CRDs to the required list, and not remove the CRDs from the Catalog bundle and remove deployment of the operators from the CSV, we'll have the OLM attempting to start new instances of the operators.

And about the local-storage-operator, I'm still not sure if we have a decision on how we make use of it. I personally believe its still the work of the underlying platform to have had the local-storage-operator deployed and configured to provide a StorageClass for OCS to use, before OCS is deployed.

**Comment by jarrpa** [comment]
> And about the local-storage-operator, I'm still not sure if we have a decision on how we make use of it. I personally believe its still the work of the underlying platform to have had the local-storage-operator deployed and configured to provide a StorageClass for OCS to use, before OCS is deployed.

I agree, but the reality is we need this functionality now and there's no other known way to get this working at this time. This is part of the design that was already approved, so once this is out of WIP I want to merge it with the dependency intact and we can remove it later once a proper solution is available.

**Comment by umangachapagain** [comment]
@jarrpa @kshlm PTAL
I have tested this and it is working well.
There are issues on local-storage-operator side, but its out of our scope.

File: deploy/olm-catalog/ocs-operator/0.0.1/ocs-operator.v0.0.1.clusterserviceversion.yaml [hunk 1]
```diff
@@ -61,6 +61,13 @@ metadata:
 spec:
   apiservicedefinitions: {}
   customresourcedefinitions:
+    required:
```
Comment (kshlm) [inline deploy/olm-catalog/ocs-operator/0.0.1/ocs-operator.v0.0.1.clusterserviceversion.yaml]: Just a nit. Let's have the owned CRDs listed first.

**Comment by umangachapagain** [comment]
@jarrpa @kshlm Resolved.

**Pull request merged**